ksqldb-api-client/src/main/java/io/confluent/ksql/api/client/impl/ConnectorDescriptionImpl.java

```diff
@@ -0,0 +1,103 @@
+/*
+ * Copyright 2021 Confluent Inc.
+ *
+ * Licensed under the Confluent Community License (the "License"); you may not use
+ * this file except in compliance with the License.  You may obtain a copy of the
+ * License at
+ *
+ * http://www.confluent.io/confluent-community-license
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OF ANY KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations under the License.
+ */
+
+package io.confluent.ksql.api.client.impl;
+
+import io.confluent.ksql.api.client.ConnectorDescription;
+import io.confluent.ksql.api.client.ConnectorType;
+import java.util.List;
+import java.util.Objects;
+
+public class ConnectorDescriptionImpl implements ConnectorDescription {
+
+  private final String connectorClass;
+  private final List<String> sources;
+  private final List<String> topics;
+  private final ConnectorType type;
+  private final String state;
+
+  public ConnectorDescriptionImpl(
+      final String connectorClass,
+      final List<String> sources,
+      final List<String> topics,
+      final ConnectorType type,
+      final String state
+
+  ) {
+    this.connectorClass = Objects.requireNonNull(connectorClass);
```

Comment:
nit: add second argument to name failure should this ever happen (and similarly for below and other classes).
```suggestion
    this.connectorClass = Objects.requireNonNull(connectorClass, "connectorClass");
```